doc/book/box/indexes.rst

```diff
@@ -24,46 +20,453 @@ The first index defined on a space is called the **primary key index**,
+    tarantool> box.space.bitset_example:insert{2,4}
+    tarantool> box.space.bitset_example:insert{3,7}
+    tarantool> box.space.bitset_example:insert{4,3}
+    tarantool> box.space.bitset_example.index.bitset:select(2, {iterator='BITS_ANY_SET'})
+
+The result will be:
+
+..  code-block:: tarantoolsession
+
+    ---
+    - - [3, 7]
+      - [4, 3]
+    ...
+
+because (7 AND 2) is not equal to 0, and (3 AND 2) is not equal to 0.
+
+Additionally, there exist
+:doc:`index iterator operations </reference/reference_lua/box_index/pairs>`.
+They can only be used with code in Lua and C/C++. Index iterators are for
+traversing indexes one key at a time, taking advantage of features that are
+specific to an index type, for example evaluating Boolean expressions when
+traversing BITSET indexes, or going in descending order when traversing TREE indexes.
+
+.. _index-box_index-operations:
+
+--------------------------------------------------------------------------------
+Index operations
+--------------------------------------------------------------------------------
+
+Index operations are automatic: if a data-manipulation request changes a tuple,
+then it also changes the index keys defined for the tuple.
+
+The simple index-creation operation that we've illustrated before is:
+
+.. cssclass:: highlight
+.. parsed-literal::
+
+    :samp:`box.space.{space-name}:create_index('{index-name}')`
+
+This creates a unique TREE index on the first field of all tuples
+(often called "Field#1"), which is assumed to be numeric.
+
+The simple SELECT request that we've illustrated before is:
+
+.. cssclass:: highlight
+.. parsed-literal::
+
+    :extsamp:`box.space.{*{space-name}*}:select({*{value}*})`
+
+This looks for a single tuple via the first index. Since the first index
+is always unique, the maximum number of returned tuples will be: one.
+You can call ``select()`` without arguments, causing all tuples to be returned.
+
+Let's continue working with the space 'tester' created in the :ref:`"Getting
+started" exercises <getting_started_db>` but first modify it:
+
+.. code-block:: tarantoolsession
+
+    tarantool> box.space.tester:format({
+             > {name = 'id', type = 'unsigned'},
+             > {name = 'band_name', type = 'string'},
+             > {name = 'year', type = 'unsigned'},
+             > {name = 'rate', type = 'unsigned', is_nullable=true}})
+    ---
+    ...
+
+Add the rate to the tuple #1 and #2:
+
+.. code-block:: tarantoolsession
+
+    tarantool> box.space.tester:update(1, {{'=', 4, 5}})
+    ---
+    - [1, 'Roxette', 1986, 5]
+    ...
+    tarantool> box.space.tester:update(2, {{'=', 4, 4}})
+    ---
+    - [2, 'Scorpions', 2015, 4]
+    ...
+
+
+And insert another tuple:
+
+.. code-block:: tarantoolsession
+
+    tarantool> box.space.tester:insert({4, 'Roxette', 2016, 3})
+    ---
+    - [4, 'Roxette', 2016, 3]
+    ...
+
+**The existing SELECT variations:**
+
+1. The search can use comparisons other than equality.
```

Comment:
Indent all contents of this list item with 4 spaces. Same for the next list items. Otherwise you're breaking the list.

Reply:
sure, I think I lost indents when moved this part from another section